scipy/optimize/zeros.c

```diff
@@ -19,24 +61,58 @@
     #define PyArgs(Operation) PyTuple_##Operation
 #endif
 
+
```

Comment:
The `ccallback_t` struct contains reservations for using Python callbacks, so there's in principle room for a bit more refactoring here.

This would also remove the need to add `ccallback_is_lowlevelcallable` etc.